### PR TITLE
Test naive implementation using a map

### PR DESCRIPTION
**Pull request by davidkarchynski**
Tried it using maps, log still overflows though

### CODE REVIEW

File: Bag.hs [hunk 1]
```diff
@@ -98,7 +98,8 @@ classifyFile f n classifyStrat = do
                         let parsedNewMessage = parseGrams wordBlackList n dlims newMessage
                         let newMessageVect = sparsifyVectSentence $ vectorizeSentence corpus parsedNewMessage
 
-                        let isSpam = classifyStrat vectSpams vectHams newMessageVect
+                        let isSpam = classSentence parsedSpams parsedHams parsedNewMessage
```
Comment (TruemanL) [inline Bag.hs]: If we do merge this in, can you not hardcode the classSentence function call here? It basically makes it so we can't use the cosine similarity strategy

**Comment by bolyachevets** [comment]
whats the reason to use Prelude.product, rather then not importing everything from the package that clashes?

**Comment by bolyachevets** [comment]
Why use concat here: nSpam = length $ concat spams ?
Would not length  spam be sufficient to find the number of sentences in spams? It seems that with concat it would find the number of grams in spams?

**Comment by bolyachevets** [comment]
is not gramOccursMap already computed in corpufier, why compute it in classifier as well? if we do 2 different maps 1 for spams, 1 for grams, there is probably some degree of duplication

**Comment by bolyachevets** [comment]
what is the role of map in your implementation. as sparse vector is already an intmap

File: Bag.hs (hunk 1 above)
Comment (davidkarchynski) [reply Bag.hs]: I will revert that before the merge, I am aware of it

**Comment by bolyachevets** [comment]
Perhaps the function computing gram map in corpufier can take advantage of spVector properties, e.g., take a look at getAllCounts in Classifier, which is doing smth similar

**Comment by bolyachevets** [comment]
Do we want classSentence code to be there if it is not used?

File: Bag.hs [hunk 2]
```diff
@@ -19,23 +19,26 @@ main :: IO ()
...
-        isSpam <- classifyFile vectSpams vectHams corpus filePath 1 strat
+        isSpam <- classifyFile vectSpams vectHams corpus filePath n strat
         let response = if (isSpam) then "This is spam" else "This is ham"
         putStrLn response
         yesNoResponse <- choiceDriver continuePrompt [] yesNoMap
-        if (yesNoResponse) then uiLoop vectSpams vectHams corpus else return ()
-         
+        if (yesNoResponse) then uiLoop vectSpams vectHams corpus n else return ()
          
+nGramsSelectPrompt = "Please select a value for n (1, 2 or 3) to use for splitting the documents into n-grams?"
```
Comment (TruemanL) [inline Bag.hs:39]: can you remove the ? at the end of this please?

**Pull request merged**